packages/account/src/Sections/Security/Passwords/deriv-email.tsx

```diff
@@ -1,68 +1,71 @@
 import React from 'react';
-import PropTypes from 'prop-types';
-import { Formik } from 'formik';
-import { withRouter } from 'react-router';
-import { WS, toTitleCase } from '@deriv/shared';
-import { Localize, localize } from '@deriv/translations';
 import { Button, Text, Input } from '@deriv/components';
+import { useVerifyEmail } from '@deriv/hooks';
+import { toTitleCase } from '@deriv/shared';
+import { observer, useStore } from '@deriv/stores';
+import { Localize, localize } from '@deriv/translations';
 import FormSubHeader from 'Components/form-sub-header';
 import SentEmailModal from 'Components/sent-email-modal';
 import UnlinkAccountModal from 'Components/unlink-account-modal';
-import { observer, useStore } from '@deriv/stores';
 
-const DerivEmail = observer(({ email, social_identity_provider, is_social_signup }) => {
-    const { common } = useStore();
-    const { is_from_derivgo } = common;
+type TDerivEmailProps = {
+    email: string;
+};
+
+/**
+ * Display the user's email address and a button to change it.
+ * @name DerivEmail
+ * @param {string} email - The user's email address.
+ * @returns {React.ReactNode}
+ */
+const DerivEmail = observer(({ email }: TDerivEmailProps) => {
+    const {
+        common: { is_from_derivgo },
+        client: { social_identity_provider, is_social_signup },
+    } = useStore();
+    const { send } = useVerifyEmail('request_email');
```

Comment:
@likhith-deriv Shouldn't we handle loading and error state too? 🤔

Reply:
We should. But I am unware of what must be the flowwhen there is an error